internal/explorer/server.go

```diff
@@ -386,19 +615,37 @@ func Setup(router *mux.Router, redisServer string, gitCommit string, database db
 		releaseVersion: gitCommit,
 	}
 
-	router.HandleFunc("/farms", mw.AsHandlerFunc(a.listFarms))
-	router.HandleFunc("/stats", mw.AsHandlerFunc(a.getStats))
-	router.HandleFunc("/nodes", mw.AsHandlerFunc(a.getNodes))
-	router.HandleFunc("/gateways", mw.AsHandlerFunc(a.getGateways))
-	router.HandleFunc("/twins", mw.AsHandlerFunc(a.listTwins))
-	router.HandleFunc("/contracts", mw.AsHandlerFunc(a.listContracts))
-	router.HandleFunc("/nodes/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getNode))
-	router.HandleFunc("/gateways/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getGateway))
-	router.HandleFunc("/nodes/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))
-	router.HandleFunc("/gateways/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))
+	// v1 endpoints
 	router.HandleFunc("/", mw.AsHandlerFunc(a.indexPage(router)))
 	router.HandleFunc("/version", mw.AsHandlerFunc(a.version))
 	router.PathPrefix("/swagger").Handler(httpSwagger.WrapHandler)
+	router.HandleFunc("/stats", mw.AsHandlerFunc(a.getV1Stats))
+
+	router.HandleFunc("/farms", mw.AsHandlerFunc(a.getV1Farms))
+	router.HandleFunc("/nodes", mw.AsHandlerFunc(a.getV1Nodes))
+	router.HandleFunc("/gateways", mw.AsHandlerFunc(a.getV1Gateways))
+	router.HandleFunc("/twins", mw.AsHandlerFunc(a.getV1Twins))
+	router.HandleFunc("/contracts", mw.AsHandlerFunc(a.getV1Contracts))
+	router.HandleFunc("/nodes/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getV1Node))
+	router.HandleFunc("/gateways/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getV1Gateway))
+	router.HandleFunc("/nodes/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))
+	router.HandleFunc("/gateways/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))
+
+	// v2 endpoints
+	router.HandleFunc("/api/v2/", mw.AsHandlerFunc(a.indexPage(router)))
+	router.HandleFunc("/api/v2/version", mw.AsHandlerFunc(a.version))
+	router.PathPrefix("/api/v2/swagger").Handler(httpSwagger.WrapHandler)
+	router.HandleFunc("/api/v2/stats", mw.AsHandlerFunc(a.getV2Stats))
+
+	router.HandleFunc("/api/v2/farms", mw.AsHandlerFunc(a.getV2Farms))
+	router.HandleFunc("/api/v2/nodes", mw.AsHandlerFunc(a.getV2Nodes))
+	router.HandleFunc("/api/v2/gateways", mw.AsHandlerFunc(a.getV2Gateways))
+	router.HandleFunc("/api/v2/twins", mw.AsHandlerFunc(a.getV2Twins))
+	router.HandleFunc("/api/v2/contracts", mw.AsHandlerFunc(a.getV2Contracts))
+	router.HandleFunc("/api/v2/nodes/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getV2Node))
+	router.HandleFunc("/api/v2/gateways/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getV2Gateway))
+	router.HandleFunc("/api/v2/nodes/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))
+	router.HandleFunc("/api/v2/gateways/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))
```

Comment:
```suggestion
	var api1 = router.PathPrefix("").Subrouter()
	var api2 = router.PathPrefix("/api/v2").Subrouter()

	// v1 endpoints
	api1.HandleFunc("/", mw.AsHandlerFunc(a.indexPage(api1)))
	api1.HandleFunc("/version", mw.AsHandlerFunc(a.version))
	api1.PathPrefix("/swagger").Handler(httpSwagger.WrapHandler)
	api1.HandleFunc("/stats", mw.AsHandlerFunc(a.getV1Stats))

	api1.HandleFunc("/farms", mw.AsHandlerFunc(a.getV1Farms))
	api1.HandleFunc("/nodes", mw.AsHandlerFunc(a.getV1Nodes))
	api1.HandleFunc("/gateways", mw.AsHandlerFunc(a.getV1Gateways))
	api1.HandleFunc("/twins", mw.AsHandlerFunc(a.getV1Twins))
	api1.HandleFunc("/contracts", mw.AsHandlerFunc(a.getV1Contracts))
	api1.HandleFunc("/nodes/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getV1Node))
	api1.HandleFunc("/gateways/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getV1Gateway))
	api1.HandleFunc("/nodes/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))
	api1.HandleFunc("/gateways/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))

	// v2 endpoints
	api2.HandleFunc("/", mw.AsHandlerFunc(a.indexPage(api2)))
	api2.HandleFunc("/version", mw.AsHandlerFunc(a.version))
	api2.PathPrefix("/swagger").Handler(httpSwagger.WrapHandler)
	api2.HandleFunc("/stats", mw.AsHandlerFunc(a.getV2Stats))

	api2.HandleFunc("/farms", mw.AsHandlerFunc(a.getV2Farms))
	api2.HandleFunc("/nodes", mw.AsHandlerFunc(a.getV2Nodes))
	api2.HandleFunc("/gateways", mw.AsHandlerFunc(a.getV2Gateways))
	api2.HandleFunc("/twins", mw.AsHandlerFunc(a.getV2Twins))
	api2.HandleFunc("/contracts", mw.AsHandlerFunc(a.getV2Contracts))
	api2.HandleFunc("/nodes/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getV2Node))
	api2.HandleFunc("/gateways/{node_id:[0-9]+}", mw.AsHandlerFunc(a.getV2Gateway))
	api2.HandleFunc("/nodes/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))
	api2.HandleFunc("/gateways/{node_id:[0-9]+}/status", mw.AsHandlerFunc(a.getNodeStatus))
```

Reply:
Subrouters can be used to create path namespaces, this would make it easier to apply middlewares independently for each subroute.
Also passing all routes without sub-routing to the index handler will result in listing v1 endpoints in the v2 homepage and vice-versa.